apps/dashboard/src/main/java/com/akto/action/CustomDataTypeAction.java

```diff
@@ -236,10 +257,82 @@ public String saveAktoDataType(){
+            List<CustomDataType> customDataTypesToBeFixed = CustomDataTypeDao.instance.findAll(Filters.eq(AktoDataType.SAMPLE_DATA_FIXED, false));
+            loggerMaker.infoAndAddToDb("Found " + customDataTypesToBeFixed.size() + " custom data types to be fixed", LogDb.DASHBOARD);
+
+            customDataTypesToBeFixed.forEach(cdt -> {
+                SingleTypeInfo.SubType st = cdt.toSubType();
+                loggerMaker.infoAndAddToDb("Dropping redacted data types for custom data type:" + cdt.getName() + ", subType:" + st.getName(), LogDb.DASHBOARD);
+                handleRedactionForSubType(st);
+                CustomDataTypeDao.instance.updateOneNoUpsert(Filters.eq(CustomDataType.NAME, cdt.getName()), Updates.set(CustomDataType.SAMPLE_DATA_FIXED, true));
+                loggerMaker.infoAndAddToDb("Dropped redacted data types for custom data type:" + cdt.getName() + ", subType:" + st.getName(), LogDb.DASHBOARD);
+            });
+
+            loggerMaker.infoAndAddToDb("Dropping redacted data types for akto data types", LogDb.DASHBOARD);
+
+            List<AktoDataType> aktoDataTypesToBeFixed = AktoDataTypeDao.instance.findAll(Filters.eq(AktoDataType.SAMPLE_DATA_FIXED, false));
+            loggerMaker.infoAndAddToDb("Found " + aktoDataTypesToBeFixed.size() + " akto data types to be fixed", LogDb.DASHBOARD);
+            aktoDataTypesToBeFixed.forEach(adt -> {
+                loggerMaker.infoAndAddToDb("Dropping redacted data types for akto data type:" + adt.getName(), LogDb.DASHBOARD);
+                SingleTypeInfo.SubType st = subTypeMap.get(adt.getName());
+                handleRedactionForSubType(st);
+                AktoDataTypeDao.instance.updateOneNoUpsert(Filters.eq(AktoDataType.NAME, adt.getName()), Updates.set(AktoDataType.SAMPLE_DATA_FIXED, true));
+                loggerMaker.infoAndAddToDb("Dropped redacted data types for akto data type:" + adt.getName(), LogDb.DASHBOARD);
+            });
+            loggerMaker.infoAndAddToDb("Dropped redacted data types successfully!", LogDb.DASHBOARD);
+        } catch (Exception e){
+            loggerMaker.errorAndAddToDb("Failed to drop redacted data types", LogDb.DASHBOARD);
+        }
+
+    }
+
+    private static void handleRedactionForSubType(SingleTypeInfo.SubType subType) {
+        loggerMaker.infoAndAddToDb("Dropping redacted data types for subType:" + subType.getName(), LogDb.DASHBOARD);
+        int skip = 0;
+        int limit = 100;
+        Set<ApiInfo.ApiInfoKey> keys = new HashSet<>();
+        while (true) {
+            List<ApiInfo.ApiInfoKey> apiInfoKeys = SingleTypeInfoDao.instance.fetchEndpointsInCollection2(subType, skip, limit);
+            loggerMaker.infoAndAddToDb("Found " + apiInfoKeys.size() + " apiInfoKeys for subType:" + subType.getName(), LogDb.DASHBOARD);
+            if(apiInfoKeys.isEmpty()){
+                break;
+            }
+            keys.addAll(apiInfoKeys);
```

Comment:
You are storing all ApiInfo in 1 set. This will go OOM. Collect a subset of URLs and execute in batches

Reply:
Fixed.